lib/devise-security/models/secure_validatable.rb

```diff
@@ -70,6 +73,15 @@ def current_equal_password_validation
         self.errors.add(:password, :equal_to_current_password) if dummy.valid_password?(password)
       end
 
+      def email_not_equal_password_validation
+        return if password.blank? || (!new_record? && !will_save_change_to_encrypted_password?)
+        dummy = self.class.new.tap do |user|
+          user.password_salt = password_salt if respond_to?(:password_salt)
+          user.password = email
```

Comment:
Upon further consideration, this requires further attention to potential uses of `case_insensitive_keys`. If email is used as case insensitive, then this validation will not be semantically correct. It would only fail validation if the password exactly matches the downcased email.
An alternative here could be to downcase `password` if `case_insensitive_keys` is set